pwiz_tools/Skyline/TestData/CommandLineTest.cs

```diff
@@ -289,6 +291,92 @@ public void ConsoleAddFastaTest()
             AssertEx.IsDocumentState(doc, 0, 2, 7, 7, 49);
         }
 
+        [TestMethod]
+        public void ConsoleNewDocumentTest()
+        {
+            TestFilesDir = new TestFilesDir(TestContext, ZIP_FILE);
+            string docPath = TestFilesDir.GetTestPath("BSA_Protea_label_free_20100323_meth3_multi.sky");
+            string outPath = TestFilesDir.GetTestPath("AddFasta_Out.sky");
+            string fastaPath = TestFilesDir.GetTestPath("sample.fasta");
+
+            // arguments that would normally be quoted on the command-line shouldn't be quoted here
+            var settings = new[]
+            {
+                "--new=" + docPath,
+                "--full-scan-precursor-isotopes=Count",
+                "--full-scan-precursor-analyzer=centroided",
+                "--full-scan-precursor-res=5",
+                "--full-scan-acquisition-method=DIA",
+                "--full-scan-isolation-scheme=All Ions",
+                "--full-scan-product-analyzer=centroided",
+                "--full-scan-product-res=5",
+                "--full-scan-rt-filter=scheduling_windows",
+                "--full-scan-rt-filter-tolerance=5",
+                "--tran-precursor-ion-charges=2,3,4",
+                "--tran-product-ion-charges=1,2",
+                "--tran-product-start-ion=" + TransitionFilter.StartFragmentFinder.ION_1.Label,
+                "--tran-product-end-ion=" + TransitionFilter.EndFragmentFinder.LAST_ION_MINUS_1.Label,
+                "--tran-product-clear-special-ions",
+                "--tran-use-dia-window-exclusion",
+                "--library-product-ions=6",
+                "--library-min-product-ions=6",
+                "--library-match-tolerance=" + 0.05 + "mz",
+                "--library-pick-product-ions=filter",
+                "--instrument-min-mz=42",
+                "--instrument-max-mz=2000",
+                "--instrument-min-time=" + 0.42,
+                "--instrument-max-time=" + 4.2,
+                "--instrument-dynamic-min-mz",
+                "--instrument-method-mz-tolerance=" + 0.42,
+                "--instrument-triggered-chromatograms",
+                "--import-fasta=" + fastaPath,
+                "--associate-proteins-group-proteins",
+                "--associate-proteins-shared-peptides=AssignedToBestProtein",
+                "--associate-proteins-minimal-protein-list",
+                "--associate-proteins-remove-subsets",
+                "--associate-proteins-min-peptides=2",
+            };
+
+            string output = RunCommand(settings);

```

Comment:
--overwrite does not currently require --new. It arguably makes sense with --out as well, although it looks like I didn't add that yet (probably meant to talk to you about it to decide whether to use it for both or just for --new). But I can see a few others I could add negative tests for.

Reply:
That would probably be fine to make it work for --out. On the other hand, it would change current functionality that certainly could break an existing pipeline. But that would change the test that causes an error when --overwrite is present to being a command-line without either --new or --out, and you would need to add a test for --out and you do need a test for --new that fails because the file exists and you didn't specify --overwrite. That seems like a pretty important test to me. Since, if that is the intended behavior it would be pretty bad if someone broke that by accident.

Reply:
It is probably worth the potentional of breaking an existing pipeline to make --new and --out consistent with respect to --overwrite. However, we definitely do not want it to be valid for just --save. That is really the entire point of --save that it saves back to the original file which exists.

Reply:
You should probably test some cases where the numbers used are outside the allowed ranges to prove that causes reasonable failure messages. Frequently there are constants that give the allowed ranges and you can make something just outside those by adding or subtracting, e.g.:

        public const double MIN_MATCH_TOLERANCE = 0.001;    // Reduced from 0.1 to 0.001 (1 ppm at 1000 m/z) for high accuracy MS/MS
        public const double MAX_MATCH_TOLERANCE = 1.0;
        public const double MAX_MATCH_TOLERANCE_PPM = 100.1;    // Increased from 1 to 100 for ppm tolerance units

You can also use the actual ARG_ properties:

ARG_NEW.ArgumentText
ARG_AP_MIN_PEPTIDES.GetArgumentTextWithValue("2")

Kind-of wish I had made that an object and used ToString() in the function body to make it easier to supply a double value.

Reply:
Check the new tests below this conversation. Do they fit the bill?